packages/blockchain-extension/test/commands/deleteExtensionDirectoryCommand.test.ts

```diff
@@ -0,0 +1,191 @@
+
+            await TestUtil.setupLocalFabric();
+        });
+
+        afterEach(() => {
+            mySandBox.restore();
+        });
+
+        it('should test the extension directory can be deleted from the command', async () => {
+            await vscode.commands.executeCommand(ExtensionCommands.DELETE_DIRECTORY);
+            reloadStub.should.have.been.calledOnce;
+            tearDownStub.should.have.been.called;
+            confirmationWarningStub.should.have.been.calledOnce;
+            removeStub.should.have.been.calledOnce;
+            environments = await FabricEnvironmentRegistry.instance().getAll();
+            environments.length.should.equal(1);
+            getAllSpy.should.have.been.calledWithExactly(true);
+        });
+
+        it('should handle an error in tearing down an environment', async () => {
+            const error: Error = new Error('some error');
+            tearDownStub.rejects(error);
+
+            await vscode.commands.executeCommand(ExtensionCommands.DELETE_DIRECTORY);
+            confirmationWarningStub.should.have.been.calledOnce;
+            removeStub.should.have.been.calledOnce;
+            reloadStub.should.have.been.calledOnce;
+            getAllSpy.should.have.been.called;
+            logSpy.should.have.not.been.called;
+        });
+
+        it('should handle an error in deleting the extension directory', async () => {
+            const error: Error = new Error('some error');
+            removeStub.rejects(error);
+
+            await vscode.commands.executeCommand(ExtensionCommands.DELETE_DIRECTORY);
+            confirmationWarningStub.should.have.been.calledOnce;
+            tearDownStub.should.have.been.called;
+            reloadStub.should.have.not.been.called;
+            pathExistsStub.should.have.been.calledOnce;
+            getAllSpy.should.have.been.called;
+            logSpy.should.have.been.calledWithExactly(LogType.ERROR, `Error deleting directory: ${error.message}`, `Error deleting directory: ${error.toString()}`);
+        });
+
+        it('should test the extension directory delete command can be cancelled', async () => {
+            confirmationWarningStub.withArgs(`This will delete the extension directory. Do you want to continue?`).resolves(false);
+            await vscode.commands.executeCommand(ExtensionCommands.DELETE_DIRECTORY);
+            confirmationWarningStub.should.have.been.called;
+            reloadStub.should.have.not.been.called;
+            tearDownStub.should.have.not.been.called;
+            tearDownStub.should.have.not.been.called;
+            removeStub.should.have.not.been.called;
+            getAllSpy.should.have.not.been.called;
+            logSpy.should.have.not.been.called;
+        });
+
+        it('should ignore any exceptions other than exceptions caused by deleting the directory', async () => {
+            getGatewayRegistryEntryStub.resolves(true);
+            getEnvironmentRegistryEntryStub.resolves(true);
```

Comment:
`getGatewayRegistryEntry` / `getEnvironmentRegistryEntryStub` don't return booleans, they return FabricGatewayRegistryEntry / FabricEnvironmentRegistryEntry, or void.

So you should give them correct return values, something like 
```
const gatewayRegEntry: FabricGatewayRegistryEntry = new FabricGatewayRegistryEntry();
gatewayRegEntry.name = '<some name>';
getGatewayRegistryEntryStub.resolves(gatewayRegEntry);


const environmentRegEntry: FabricEnvironmentRegistryEntry = new FabricEnvironmentRegistryEntry();
environmentRegEntry.name = '<some name>';
getEnvironmentRegistryEntryStub.resolves(environmentRegEntry);
```